ocs_ci/ocs/registry.py

```diff
@@ -0,0 +1,282 @@
+
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.resources import pod
+from ocs_ci.utility.utils import run_cmd
+from tests import helpers
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.framework import config
+
+
+logger = logging.getLogger(__name__)
+
+
+def change_registry_backend_to_ocs():
+    """
+    Function to deploy registry with OCS backend.
+
+    Raises:
+        AssertionError: When failure in change of registry backend to OCS
+    """
+    pv_obj = helpers.create_pvc(
+        sc_name=constants.DEFAULT_SC_CEPHFS, pvc_name='registry-cephfs-rwx-pvc',
+        namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE, size='100Gi',
+        access_mode=constants.ACCESS_MODE_RWX
+    )
+    helpers.wait_for_resource_state(pv_obj, 'Bound')
+    ocp_obj = ocp.OCP(
+        kind=constants.CONFIG, namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+    )
+    param_cmd = f'[{{"op": "add", "path": "/spec/storage", "value": {{"pvc": {{"claim": "{pv_obj.name}"}}}}}}]'
+    assert ocp_obj.patch(
+        resource_name=constants.IMAGE_REGISTRY_RESOURCE_NAME, params=param_cmd
+    ), f"Registry pod storage backend to OCS is not success"
+
+    # Validate registry pod status
+    validate_registry_pod_status()
+
+    # Validate pvc mount in the registry pod
+    validate_pvc_mount_on_registry_pod()
+
+
+def get_registry_pod_obj():
+    """
+    Function to get registry pod obj
+
+    Returns:
+        pod_obj (obj): Registry pod obj
+
+    Raises:
+        UnexpectedBehaviour: When image-registry pod is not present.
+    """
+    # Sometimes when there is a update in config crd, there will be 2 registry pods
+    # i.e. old pod will be terminated and new pod will be up based on new crd
+    # so below loop waits till old pod terminates
+    wait_time = 30
+    iteration = 0
+    while True:
+        iteration += 1
+        pod_data = pod.get_pods_having_label(
+            label='docker-registry=default', namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+        )
+        pod_obj = [pod.Pod(**data) for data in pod_data]
+        if len(pod_obj) == 1:
+            break
+        elif len(pod_obj) == 0:
+            raise UnexpectedBehaviour("Image-registry pod not present")
+        elif iteration > 6:
+            raise UnexpectedBehaviour("Waited for 3 mins Image-registry pod is not in Running state")
+        else:
+            logger.info(f"Waiting for 30 sec's for registry pod to be up iteration {iteration}")
+            time.sleep(wait_time)
+    return pod_obj
+
+
+def get_oc_podman_login_cmd():
+    """
+    Function to get oc and podman login commands on node
+
+    Returns:
+        cmd_list (list): List of cmd for oc/podman login
+    """
+    user = config.RUN['username']
+    filename = os.path.join(
+        config.ENV_DATA['cluster_path'],
+        config.RUN['password_location']
+    )
+    with open(filename) as f:
+        password = f.read()
+    helpers.refresh_oc_login_connection()
```

Comment:
Didn't get why we need to do this?  Once we execute the run-ci we have exported kubeconfig so why this is needed at all?

Reply:
Observed problems to get_user_token if login is not refreshed. due to this podman login will fail.

Reply:
So is this needed just to obtain the token?

Cannot we use this approach:
https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/

```
APISERVER=$(oc config view --minify -o jsonpath='{.clusters[0].cluster.server}')
SECRET_NAME=$(oc get serviceaccount default -o jsonpath='{.secrets[0].name}')
TOKEN=$(oc get secret $SECRET_NAME -o jsonpath='{.data.token}' | base64 --decode)
```

I tried above and then:
`curl $APISERVER/api --header "Authorization: Bearer $TOKEN" --insecure`
worked for me.

Reply:
Will test the code with the changes and update sooner.

Reply:
In first place I am not sure if this token is that one we really need :) just when I looked to google how to obtain the token that was what I found but not sure if this is the same token as we can see from whoami command.  If you can confirm that this token is what we need it can be part of that function but I hope that maybe @fbalak can shed some light here as git blame wrote that he wrote that function. So maybe @fbalak can lear us something more about this.

Reply:
@petr-balogh This is different token. I tried token from https://github.com/red-hat-storage/ocs-ci/pull/868/files/2387da9a0b97fd8fcf81332556059a475086bfa6#r334872621 generated from secret as you wrote but it didn't work when I tried to use it with prometheus API.

Reply:
With the $TOKEN podman login failed.
```
podman login default-route-openshift-image-registry.apps.rperiyas-cls.qe.rh-ocs.com -u kubeadmin -p ZXlKaGJHY2lPaUpTVXpJMU5pSXNJbXRwWkNJNklpSjkuZXlKcGMzTWlPaUpyZFdKbGNtNWxkR1Z6TDNObGNuWnBZMlZoWTJOdmRXNTBJaXdpYTNWaVpYSnVaWFJsY3k1cGJ5OXpaWEoyYVdObFlXTmpiM1Z1ZEM5dVlXMWxjM0JoWTJVaU9pSmtaV1poZFd4MElpd2lhM1ZpWlhKdVpYUmxjeTVwYnk5elpYSjJhV05sWVdOamIzVnVkQzl6WldOeVpYUXVibUZ0WlNJNkltUmxabUYxYkhRdGRHOXJaVzR0Y3pkaU5Hc2lMQ0pyZFdKbGNtNWxkR1Z6TG1sdkwzTmxjblpwWTJWaFkyTnZkVzUwTDNObGNuWnBZMlV0WVdOamIzVnVkQzV1WVcxbElqb2laR1ZtWVhWc2RDSXNJbXQxWW1WeWJtVjBaWE11YVc4dmMyVnlkbWxqWldGalkyOTFiblF2YzJWeWRtbGpaUzFoWTJOdmRXNTBMblZwWkNJNklqRTBZekl3WlRGaUxXVm1ORFV0TVRGbE9TMDRPVFprTFRBeU16SXpOR0k1WmpFeU9DSXNJbk4xWWlJNkluTjVjM1JsYlRwelpYSjJhV05sWVdOamIzVnVkRHBrWldaaGRXeDBPbVJsWm1GMWJIUWlmUS5CNGF3RGJ0VC0xeHRzVnlqSGtTRFhGUzF5R0kxTE1vdmk2bEdJN284d1FlZXRIaFJUT2FVWjFDaHpFbFA4X0tNTVlSRlE2RTh0UVZLdDdFTVVXSFV6bHZVaXNzSmprMjBaQ0VyUS1KejNpZTd4ZWpuVDVQa0g1S2k1WlZpay0xUkVJOXpVMGxFWmo5alNHeU9mU21UUWhLaEt4Qm9Ja1FvbG41bEJ2VkpydnFNcXdGOUI2MFBJSzlZSTR3VDBsQ3dLVU5aZ0VBNzhIbEsxZnB0dTZfeFhkUnhydEYteFdwZXQ1Tk81UDkxZUx2d1VIbnBMNW5TTFFkelhDYnNMNEdwRlN3Sll6M3ZPSlhUcWcyWGJOUUtsZWhMSTNHQUJGVE9fZktFMnBkZUF3QmhxLVRjNjNTTUhyUUdHdGFjeXVUUkwwTjVESm02LXVMVVhUalB4NXVhelE=

Error: error logging into "default-route-openshift-image-registry.apps.rperiyas-cls.qe.rh-ocs.com": invalid username/password
'''